curandommatrices/src/circular.jl

```diff
@@ -0,0 +1,61 @@
+# FIXME: this can be accelerated
+function cplx_phase!(a)
+    i = (blockIdx().x-1) * blockDim().x + threadIdx().x
+    if i <= length(a)
+        @inbounds a[i] = a[i] / sqrt(real(a[i])^2 + imag(a[i])^2)
+    end
+    return nothing
+end
+
+# FIXME: use blocks to support larger than 1024
+
+function _qr_fix!(z::CuMatrix)
+    q, r = CuArrays.qr!(z)
+    ph = diag(r)
+    len = min(length(ph), 1024)
```

Comment:
Why the magic number `1024` ? Explain it or at least point to some information.

Reply:
This is just an ugly hack as `warpsize()` results in a segfault.

Reply:
Then document it.